code/game/mecha/mecha.dm

```diff
@@ -551,7 +551,8 @@
 					step(obstacle, dir)
 			else if(istype(obstacle, /mob))
 				step(obstacle, dir)
-
+	if(istype(obstacle, /obj/structure/table) || istype(obstacle, /obj/structure/rack))

```

Comment:
probably refering to putting istype()s on Bump() which is snowflake, specially when you're already handling it on melee attack